src/test/java/net/snowflake/client/jdbc/structuredtypes/ResultSetStructuredTypesGetStringGetBytesCompatibilityLatestIT.java

```diff
@@ -0,0 +1,116 @@
+@RunWith(Parameterized.class)
+@Category(TestCategoryResultSet.class)
+public class ResultSetStructuredTypesGetStringGetBytesCompatibilityLatestIT
+    extends ResultSetStructuredTypesBaseIT {
+
+  private static Map<ResultSetFormatType, Connection> connections = new HashMap<>();
+
+  @Before
+  public void setUpConnection() throws SQLException {
+    // We init connection here since we need to set server properties that cannot be set in GH
+    // actions and before class is running even when all the tests have conditional ignore of tests
+    Connection connection = connections.get(queryResultFormat);
+    if (connection == null) {
+      connections.put(queryResultFormat, initConnection(queryResultFormat));
+    }
+  }
+
+  @AfterClass
+  public static void closeConnections() throws SQLException {
+    for (Connection connection : connections.values()) {
+      connection.close();
+    }
+  }
+
+  @Parameterized.Parameters(name = "format={0},sql={1}")
+  public static Collection<Object[]> data() {
+    Map<String, String> samples = new LinkedHashMap<>();
+    samples.put("select {'a':3}::map(text, int);", "{\"a\":3}");
+    samples.put("select {'a':'bla'}::map(text, text);", "{\"a\":\"bla\"}");
+    samples.put("select {'1':'bla'}::map(int, text);", "{\"1\":\"bla\"}");
+    samples.put("select {'1':[1,2,3]}::map(int, ARRAY(int));", "{\"1\":[1,2,3]}");
+    samples.put(
+        "select {'1':{'string':'a'}}::map(int, OBJECT(string VARCHAR));",
+        "{\"1\":{\"string\":\"a\"}}");
+    samples.put(
+        "select {'1':{'string':'a'}}::map(int, map(string, string));",
+        "{\"1\":{\"string\":\"a\"}}");
+    samples.put(
+        "select {'1':[{'string':'a'},{'bla':'ble'}]}::map(int, array(map(string, string)));",
+        "{\"1\":[{\"string\":\"a\"},{\"bla\":\"ble\"}]}");
+    samples.put("select [1,2,3]::array(int)", "[1,2,3]");
+    samples.put(
+        "select [{'a':'a'}, {'b':'b'}]::array(map(string, string))",
+        "[{\"a\":\"a\"}, {\"b\":\"b\"}]");
+    samples.put(
+        "select [{'string':'a'}, {'string':'b'}]::array(object(string varchar))",
+        "[{\"string\":\"a\"}, {\"string\":\"b\"}]");
+    samples.put("select {'string':'a'}::object(string varchar)", "{\"string\":\"a\"}");
+    samples.put("select {'string':[1,2,3]}::object(string array(int))", "{\"string\":[1,2,3]}");
+    samples.put(
+        "select {'string':{'a':15}}::object(string object(a int))", "{\"string\":{\"a\":15}}");
+    samples.put(
+        "select {'string':{'a':15}}::object(string map(string,int))", "{\"string\":{\"a\":15}}");
+    samples.put(
+        "select {'string':{'a':{'b':15}}}::object(string object(a map(string, int)))",
+        "{\"string\":{\"a\":{\"b\":15}}}");
+    samples.put(
+        "select {'string':{'a':{'b':[{'c': 15}]}}}::object(string map(string, object(b array(object(c int)))))",
+        "{\"string\":{\"a\":{\"b\":[{\"c\":15}]}}}");
+    samples.put("select [1,2,3]::VECTOR(INT, 3)", "[1,2,3]");
```

Comment:
Is it enough testing here only string and int type?

Reply:
Vector type does not support more currently https://docs.snowflake.com/en/sql-reference/data-types-vector